src/hooks/useReportWithTransactionsAndViolations.ts

```diff
@@ -7,18 +7,18 @@ import type {Report, Transaction, TransactionViolation} from '@src/types/onyx';
 
 function useReportWithTransactionsAndViolations(reportID?: string): [OnyxEntry<Report>, Transaction[], OnyxCollection<TransactionViolation[]>] {
     const [report] = useOnyx(`${ONYXKEYS.COLLECTION.REPORT}${reportID ?? CONST.DEFAULT_NUMBER_ID}`);
-    const [transactions = []] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION, {
+    const [transactions] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION, {
         selector: (_transactions) => reportTransactionsSelector(_transactions, reportID),
     });
     const [violations] = useOnyx(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, {
         selector: (allViolations) =>
             Object.fromEntries(
                 Object.entries(allViolations ?? {}).filter(([key]) =>
-                    transactions.some((transaction) => transaction.transactionID === key.replace(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, '')),
+                    transactions?.some((transaction) => transaction.transactionID === key.replace(ONYXKEYS.COLLECTION.TRANSACTION_VIOLATIONS, '')),
                 ),
             ),
     });
-    return [report, transactions, violations];
+    return [report, transactions ?? [], violations ?? {}];
```

Comment:
can we memoize the `transactions ?? []` and `violations ?? {}` so the reference to the default values is stable when `transactions` or `violations` is undefined/null?

I'm asking for this because we have code using `useMemo` and passing as a dependency these values. If the `[]` / `{}` is recreated on each render, then those `useMemo` that watch these values won't be able to cache properly.

For example:

https://github.com/Expensify/App/blob/db79b1478eaa5b406de4189d978152444b02cb6d/src/components/ReportActionItem/ReportPreview.tsx#L166-L176

Reply:
an alternative could be to define some constants outside the component for the defaults outside, so they stay as the same instance always:

```
const DEFAULT_TRANSACTIONS = [];
const DEFAULT_VIOLATIONS = {};
```

then in the component:

```
    return [report, transactions ?? DEFAULT_TRANSACTIONS, violations ?? DEFAULT_VIOLATIONS];
```

Reply:
thanks @aldo-expensify , I went with defining the defaults outside, tested it and it works properly 👍

Reply:
Thanks for addressing